src/derivative.jl

```diff
@@ -0,0 +1,56 @@
+"""
+    struct DerivativeFilter <: AbstractRadIIRFilter
+
+Working example:
+```julia
+using LegendDSP
+using Unitful
+
+signal = rand(100)
+t = range(0u"ms", 20u"ms", 100)
+
+wf = RDWaveform(t, signal)
+flt = DerivativeFilter()
+wf_new = flt(wf)
+```
+Constructors:
+
+* ```$(FUNCTIONNAME)(; fields...)```
+
+Fields:
+
+$(TYPEDFIELDS)
+"""
+Base.@kwdef struct DerivativeFilter{T <: RadiationDetectorDSP.RealQuantity} <: AbstractRadIIRFilter
+    "Filter gain"
+    gain::T = 1
+end
+export DerivativeFilter
+
+struct DerivativeFilterInstance{T <: RadiationDetectorDSP.RealQuantity, TG <: RadiationDetectorDSP.RealQuantity} <: AbstractRadSigFilterInstance{LinearFiltering}
+    gain::TG
+    n_input::Int
+end
```

Comment:
I'm not sure that the parametric types (`T` and `TG`) are needed for, I just put them there.